apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -308,40 +297,77 @@ private static void addJmxMetricRegistration(final JmxMetric jmxMetric, List<Jmx
                     }
-                } catch (AttributeNotFoundException e) {
-                    logger.warn("Can't create metric '{}' because attribute '{}' could not be found", jmxMetric, attribute.getJmxAttributeName());
+                } else {
+                    final Object value = server.getAttribute(objectName, attribute.getJmxAttributeName());
+                    try {
+                        addJmxMetricRegistration(jmxMetric, registrations, objectName, value, attribute, attribute.getJmxAttributeName(), null);
+                    } catch (AttributeNotFoundException e) {
+                        logger.warn("Can't create metric '{}' because attribute '{}' could not be found", jmxMetric, attribute.getJmxAttributeName());
+                    }
                 }
             }
         }
     }
 
+    private static String metricPrepend(Labels labels) {
+        List<String> keys = labels.getKeys();
+        for (int i = 0; i < keys.size(); i++) {
+            if ("type".equals(keys.get(i))) {
+                return labels.getValue(i)+".";
+            }
+        }
+        return "";
+    }
+
+    private static void addJmxMetricRegistration(JmxMetric jmxMetric, List<JmxMetricRegistration> registrations, ObjectName objectName, Object value, JmxMetric.Attribute attribute, String attributeName, String metricPrepend) throws AttributeNotFoundException {
+        if (value instanceof Number) {
+            logger.debug("Found number attribute {}={}", attribute.getJmxAttributeName(), value);
+            registrations.add(new JmxMetricRegistration(attribute.getMetricName(metricPrepend == null? attributeName : metricPrepend+attributeName),
+                    attribute.getLabels(objectName),
+                    attributeName,
+                    null,
+                    objectName));
+        } else if (value instanceof CompositeData) {
+            final CompositeData compositeValue = (CompositeData) value;
+            for (final String key : compositeValue.getCompositeType().keySet()) {
+                if (compositeValue.get(key) instanceof Number) {
+                    logger.debug("Found composite number attribute {}.{}={}", attribute.getJmxAttributeName(), key, value);
+                    registrations.add(new JmxMetricRegistration(attribute.getCompositeMetricName(key, metricPrepend == null? attributeName : metricPrepend+attributeName),
+                            attribute.getLabels(objectName),
+                            attributeName,
+                            key,
+                            objectName));
+                } else {
+                    if (!"*".equals(attribute.getJmxAttributeName())) {
+                        logger.warn("Can't create metric '{}' because composite value '{}' is not a number: '{}'", jmxMetric, key, value);
+                    }
+                }
+            }
+        } else {
+            if (!"*".equals(attribute.getJmxAttributeName())) {
```

Comment:
[minor] we could probably factorize this by adding `isWildCard` to `JmxMetric.Attribute` class for the 3 instances of this expression.

Reply:
thanks, yes config lost after a reset. Fixed in the commit